packages/blade/src/components/Accordion/AccordionButton.web.tsx

```diff
@@ -0,0 +1,60 @@
+import type { ReactElement } from 'react';
+import { StyledAccordionButton } from './StyledAccordionButton';
+import type { AccordionButtonProps } from './types';
+import { BaseBox } from '~components/Box/BaseBox';
+import { makeAccessible } from '~utils';
+import { Heading } from '~components/Typography';
+import { ChevronDownIcon } from '~components/Icons';
+import { useCollapsible } from '~components/Collapsible/CollapsibleContext';
+
+const AccordionButton = ({ index, icon: Icon, children }: AccordionButtonProps): ReactElement => {
+  const { setIsExpanded } = useCollapsible();
+
+  const onClick = (): void => setIsExpanded((prev) => !prev);
+
+  const _index =
+    typeof index === 'number' ? (
+      <Heading size="small" marginRight="spacing.2">
+        {index + 1}.
+      </Heading>
+    ) : null;
+
+  const _icon = Icon && (
+    <Icon size="medium" color="currentColor" marginRight="spacing.3" marginY="spacing.2" />
+  );
+
+  if (_index && _icon) {
+    console.warn(`[Blade: Accordion]: showNumberPrefix and icon shouldn't be used together`);
```

Comment:
Right, intentionally didn't throw an error because I'm afraid people might set some prop values based on API responses and miss out testing all edge cases. In this case this might not get caught in development 😨

Reply:
Related #272

Reply:
that should be application's headache, otherwise we are just giving cushion to people for shipping with subpar testing 😢

Reply:
also i feel showing bad ux is better than not showing at all